tools/rpkg/src/duckdbr.cpp

```diff
@@ -214,29 +215,27 @@ SEXP duckdb_prepare_R(SEXP connsexp, SEXP querysexp) {
 	auto stmtholder = new RStatement();
 	stmtholder->stmt = move(stmt);
 
-	SEXP stmtsexp = PROTECT(R_MakeExternalPtr(stmtholder, R_NilValue, R_NilValue));
-	R_RegisterCFinalizer(stmtsexp, (void (*)(SEXP))duckdb_finalize_statement_R);
-
 	SEXP retlist = PROTECT(NEW_LIST(6));
 	if (!retlist) {
-		UNPROTECT(2); // retlist, stmtsexp
+		UNPROTECT(1); // retlist
```

Comment:
Can we make `NEW_*()` throw an error to simplify the code here and everywhere?

Reply:
NEW_* are R API macros and we have no control over their behaviour